release.xsh

```diff
@@ -83,16 +118,102 @@ def pipify():
     ./setup.py sdist upload
 
 
-def condaify(ver):
+def shatar(org, repo, target):
+    """Returns the SHA-256 sum of the {ver}.tar.gz archive from github."""
+    oldpwd = $PWD
+    cd /tmp
+    url = "https://github.com/{0}/{1}/archive/{2}.tar.gz"
+    url = url.format(org, repo, target)
+    curl -L -O @(url)
+    sha, _ = $(sha256sum @('{}.tar.gz'.format(target))).split()
+    cd @(oldpwd)
+    return sha
+
+
+def feedstok_repos(ghuser):
+    """Returns the origin and upstream repo URLs for the feedstock."""
+    origin = 'git@github.com:{ghuser}/{feedstock}.git'
+    origin = origin.format(ghuser=ghuser, feedstock=FEEDSTOCK_REPO)
+    upstream = 'git@github.com:conda-forge/{feedstock}.git'
+    upstream = upstream.format(feedstock=FEEDSTOCK_REPO)
+    return origin, upstream
+
+
+def condaify(ver, ghuser):
     """Make and upload conda packages."""
-    conda_dir = os.path.dirname(os.path.dirname($(which conda)))
-    conda_bld = os.path.join(conda_dir, 'conda-bld')
-    rm -f @(os.path.join(conda_bld, 'src_cache', 'xonsh.tar.gz'))
-    conda build --no-test recipe
-    pkgpath = os.path.join(conda_bld, '*', 'xonsh-{0}*.tar.bz2'.format(ver))
-    pkg = __xonsh_glob__(pkgpath)[0]
-    conda convert -p all -o @(conda_bld) @(pkg)
-    anaconda upload -u xonsh @(__xonsh_glob__(pkgpath))
+    origin, upstream = feedstock_repos(ghuser)
+    if not os.path.isdir('feedstock'):
+        git clone @(origin) feedstock
+    # make sure master feedstock is up to date
+    cd feedstock
+    git checkout master
+    git pull @(upstream) master
+    # make and modify version branch
+    git checkout -b @(ver) master or git checkout @(ver)
+    cd recipe
+    set_ver = '{% set version = "' + ver + '" %}'
+    set_sha = '  sha256: ' + shatar(UPSTREAM_ORG, UPSTREAM_REPO, ver)
+    replace_in_file('{% set version = ".*" %}', set_ver, 'meta.yaml')
+    replace_in_file('\s+sha256:.*', set_sha, 'meta.yaml')
+    cd ..
+    git commit -am @("updated v" + ver)
+    git push --set-upstream @(origin) @(ver)
+    cd ..
+    if github3 is not None:
+        open_feedstock_pr(ver, ghuser)
+
+
+def create_ghuser_token(ghuser, credfile):
+    """Acquires a github token, writes a credentials file, and returns
+    the token.
+    """
+    password = ''
+    while not password:
+        password = getpass('GitHub Password for {0}: '.format(ghuser))
+    note = 'github3.py release.xsh ' + PROJECT + ' ' + socket.gethostname()
+    note_url = PROJECT_URL
+    scopes = ['user', 'repo']
+    auth = github3.authorize(ghuser, password, scopes, note, note_url)
```

Comment:
this should probably support 2FA
http://github3py.readthedocs.io/en/master/examples/two_factor_auth.html